### ISSUE TITLE
Variable is assigned but never read - false positive with DO x TO

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#60]
The following code creates a false positive for the Unused variable rule
(The content of a variable is written during the flow of execution, but never read)

```
DEFINE VARIABLE liCount AS INTEGER NO-UNDO.
DO liCount = 1 TO 100:
    /* FOO */
END.
```

**Comment by gquerret** [comment]
Technically, the variable is never read and should report an issue, but the issue isn't valuable.
Will work on adding an exception in the rule.

**Comment by gquerret** [comment]
> Technically, the variable is never read and should report an issue
That's not true in fact, as the variable has to be read (even if it's internal to AVM) to execute the comparison and exit the `DO` loop.

BlockIterator tree parser statement changed to reflect that variable is both read and written.
Will work on both `DO` and `REPEAT` loops.

**Comment by gquerret** [comment]
Fixed in master - Test case added in riverside-rules
